File: C++/13. Vector/Project1/Source.cpp
#include "Maths.h"

int main()
{
	Vector3D<float> A(3, 2, 5);
	Vector3D<float> B(7, 11, 2);

	std::cout << Dot(A, B) << std::endl;
	A.Normalize().Print();
	Cross(A, B).Print();
	std::cout << A.Magnitude() << std::endl;
	(A + B).Print();
	Lerp(A, B, 0.5).Print();

	std::cout << std::endl;

	Vector4D<float> Color(0x564E45);
	
	Color.Print();
	Color.Normalize().Print();

	std::cout << std::endl;

	std::cout << Maths::Lerp(2, 5, 0.5) << std::endl;
	std::cout << Maths::ToDegrees(PI / 2) << std::endl;
	std::cout << Maths::ToRadians(90.0f) << std::endl;

	std::cout << std::endl;
	system("pause");
	return 0;
}

File: Python/5. VectorMath/source.py
from Maths import *
from os import system

def main():
	system('cls')
	
	print("Vector2D")
	print("----------------------------------------")
	
	A = Vector2D(5, 5)
	B = Vector2D(3, 1)
	
	print("A =", A, "B =", B)
	print("Dot Product:", Dot(A, B))
	print("Nomilization of A:", A.Normalize())
	print("Cross Product:", Cross(A, B))
	print("Magnitude of A:", A.Magnitude())
	
	print()
	print("Arithmatic:")
	print("A + B:", A + B)
	print("A - B:", A - B)
	print("A * B:", A * B)
	print("A / B:", A / B)
	
	print()
	system('pause')
	system('cls')
	
	print("Vector3D")
	print("----------------------------------------")
	
	A = Vector3D(5, 5, 5)
	B = Vector3D(3, 1, 2)
	
	print("A =", A, "B =", B)
	print("Dot Product:", Dot(A, B))
	print("Nomilization of A:", A.Normalize())
	print("Cross Product:", Cross(A, B))
	print("Magnitude of A:", A.Magnitude())
	
	print()
	print("Arithmatic:")
	print("A + B:", A + B)
	print("A - B:", A - B)
	print("A * B:", A * B)
	print("A / B:", A / B)
	
	print()
	system('pause')
	system('cls')
	
	print("Vector4D")
	print("----------------------------------------")
	
	A = Vector4D(5, 5, 5, 5)
	
	print("A =", A)
	print("Nomilization of A:", A.Normalize())
	
	Color = hex(0x34EA4AFF)
	A = Vector4D(Color)
	
	print("A constructed from " + Color + ":", A)
	
	print()
	system('pause')
	system('cls')
	
	
main()